src/pydvl/value/shapley/classwise.py

```diff
@@ -0,0 +1,301 @@
+"""
+Refs:
+
+[1] CS-Shapley: Class-wise Shapley Values for Data Valuation in Classification (https://arxiv.org/abs/2211.06800)
+"""
+
+# TODO rename function, and object refs, when transferred to pyDVL.
+
+import numbers
+import operator
+from functools import reduce
+from typing import Optional, Sequence, Tuple
+
+import numpy as np
+from numpy._typing import NDArray
+
+from pydvl.utils import MapReduceJob, ParallelConfig, Scorer, SupervisedModel, Utility
+
+__all__ = [
+    "class_wise_shapley",
+    "CSScorer",
+]
+
+from sklearn.metrics import accuracy_score
+from tqdm import tqdm
+
+from pydvl.utils.numeric import random_powerset_group_conditional
+from pydvl.value import StoppingCriterion, ValuationResult
+
+IntArray = NDArray[np.int_]
```

Comment:
Although convenient, this type definition is not used anywhere else in the code. The inconsistency will make this code harder to read, so I would remove this, or change it consistently everywhere in a separate PR.